src/renderers/dom/client/ReactBrowserEventEmitter.js

```diff
@@ -234,13 +251,37 @@ var ReactBrowserEventEmitter = Object.assign({}, ReactEventEmitterMixin, {
    *
    * @param {string} registrationName Name of listener (e.g. `onClick`).
    * @param {object} contentDocumentHandle Document which owns the container
+   * @return {function} A function that removes ALL listeners that were added.
    */
   listenTo: function(registrationName, contentDocumentHandle) {
     var mountAt = contentDocumentHandle;
     var isListening = getListeningForDocument(mountAt);
     var dependencies =
       EventPluginRegistry.registrationNameDependencies[registrationName];
 
+    // aggregates handler remove functions...
+    var listenerRemovers = [];
+    var trapBubbledEvent = function(topLevelType, handlerBaseName) {
+      var remover = ReactBrowserEventEmitter.ReactEventListener.trapBubbledEvent(
+        topLevelType,
+        handlerBaseName,
+        mountAt
+      );
+      if (remover && remover.remove) {
+        listenerRemovers.push(remover.remove);
+      }
+    };
+    var trapCapturedEvent = function(topLevelType, handlerBaseName, handle) {
+      var remover = ReactBrowserEventEmitter.ReactEventListener.trapCapturedEvent(
+        topLevelType,
+        handlerBaseName,
+        handle
+      );
+      if (remover && remover.remove) {
+        listenerRemovers.push(remover.remove);
+      }
+    };
```

Comment:
not sure if the names $trapBubbledEvent and $trapCapturedEvent should be different so theres no confusing about them being refs to the ones that live in ReactEventListener.